docs/proposals/api_design.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "HTTP API Design"
+linkTitle: "HTTP API Design"
+weight: 1
+slug: http-api-design
+---
+
+# HTTP API Design
+
+## Overview
+
+The purpose of this design document is to propose a set of standards that should be the basis of the Cortex HTTP API. This document will outline the current state of the Cortex http api and describe limitations that result from the current approach. It will also outline a set of paradigms on how http routes should be created within Cortex.
+
+## Current Design
+
+As things currently stand, the majority of HTTP API calls exist under the `/api/prom` path prefix. This prefix is configurable. However, since this prefix is shared between all the modules which leads to conflicts if the Alertmanager is attempted to be run as as part of the single binary.
+
+## Proposed Design
+
+### Module-Based Routing
+
+Cortex incorporates three separate APIs: Alertmanager, Prometheus, and Cortex. Each of these APIs should use a separate route prefix that accurately describes the API. Currently, all of the api calls in Cortex reside under the configured http prefix. Instead the following routing tree is proposed:
+
+#### `/prometheus/*`
+
+Under this path prefix, Cortex will act as a Prometheus web server. It will host all of the required Prometheus api endpoints. For example to query cortex the endpoint `/prometheus/api/v1/query_range` will be used.
+
+#### `/alertmanager/*`
+
+Under this path prefix, Cortex will act as a Alertmanager web server. In this case, it will forward requests to the alertmanager and support the alertmanager API. This means for a user to access their Alertmanager UI, they will use the `/alertmanager` path of cortex.
+
+#### `/api/v1/*` -- The cortex API will exist under this path prefix.
+
+- `/push` & `/read` -- The Prometheus remote write endpoints will be fulfilled directly under the `/api/v1/*` path
```

Comment:
`/read` is a Prometheus API endpoint.

Reply:
@gouthamve I noticed. It's a bit strange since we currently host the read endpoint under `/api/prom/read`. It should be `/api/prom/api/v1/read`. In this case it make sense to move both of these endpoints under the Prometheus API:

`/prometheus/api/v1/read` & `/prometheus/api/v1/write` WDYT?